build/Dockerfile.ubi

```diff
@@ -0,0 +1,16 @@
+ARG binary_name=kfctl
```

Comment:
Since the change is more about OpenDataHub and with this `Dockerfile.ubi`, I wonder whether we should just leave the other `Dockerfile` as is. That is `Dockerfile.ubi` can be used to build both kubeflow and opendatahub operator images while `Dockerfile` is purely for kubeflow operator image.

Reply:
Ah, sure, we can. I just wanted to make sure things will behave consistently, but whatever is preferred